backend/open_webui/utils/oauth.py

```
@@ -200,24 +233,59 @@ async def handle_callback(self, provider, request, response):
         if not email:
-            log.warning(f"OAuth callback failed, email is missing: {user_data}")
-            raise HTTPException(400, detail=ERROR_MESSAGES.INVALID_CRED)
+            # If the provider is GitHub,and public email is not provided, we can use the access token to fetch the user's email
+            if provider == "github":
+                try:
+                    access_token = token.get("access_token")
+                    headers = {"Authorization": f"Bearer {access_token}"}
+                    async with aiohttp.ClientSession() as session:
+                        async with session.get(
+                            "https://api.github.com/user/emails", headers=headers
+                        ) as resp:
```

Ruff (0.8.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)